frontend/src/hooks/Gitea.js

```diff
@@ -38,6 +39,20 @@ const fetcher = url =>
     headers: { Accept: 'application/json' },
   }).then(r => r.json())
 
+/**
+ * Fetch repos then flatten it into projects, i.e., each multi project as a standalone project.
+ * @param {string} url
+ * @returns
+ */
+const fetchThenFlatten = url =>
+  fetch(url, {
+    method: 'GET',
+    mode: 'cors',
+    headers: { Accept: 'application/json' },
+  })
+    .then(r => r.json())
+    .then(repos => getFlatProjects(repos))
+
```

Comment:
Why is this being done in a custom fetcher here? Also, haven't looked closely enough but this seems strange to me: why is there a `useUserRepos` and a `getUserRepos` and a `getFlatProjects` all defined seperately? Seems like you could just have a `getProjects` that can do `getProjects(user)` and a hook that makes use of it.

Reply:
`getFlatProjects` takes an array of repos, and returns a  Promise of projects, these repos can come from different sources:
1. In the `[usename]` page, on the server
https://github.com/kitspace/kitspace-v2/blob/315842ee420361e9a31f4e9e9a7aaf55e1ffa05e/frontend/src/pages/%5Busername%5D/index.js#L13-L15
2. In the '/search'  page, on the server from
   - a search query 
    https://github.com/kitspace/kitspace-v2/blob/315842ee420361e9a31f4e9e9a7aaf55e1ffa05e/frontend/src/pages/search.js#L21
   - or all repos
   https://github.com/kitspace/kitspace-v2/blob/315842ee420361e9a31f4e9e9a7aaf55e1ffa05e/frontend/src/pages/search.js#L28

I got rid of the custom fetcher and improve variables names in 3df01bd.